resolwe/flow/views.py

```diff
@@ -219,18 +219,30 @@ class ResolwePermissionsMixin(object):
     """
 
     def _fetch_user(self, query):
-        """Get user by ``pk`` or ``email``. Return ``None`` if doesn't exist."""
+        """Get user by ``pk`` or ``username``. Return ``None`` if doesn't exist."""
+        User = get_user_model()
+
         try:
```

Comment:
Too many queries and too complicated. Maybe define a `get_by_id_or_username(value)` method on the User's object manager. If the value is int, query by id, else by the username.